File: modules/widget/src/WidgetModule.ts
import { NativeModule, requireNativeModule } from 'expo';
import { WidgetModuleEvents } from './Widget.types';

declare class WidgetModule extends NativeModule<WidgetModuleEvents> {
    getItem: (...args: string[]) => '';
    setItem: (...args: string[]) => {};
    setImage: (...args: string[]) => '';
    getImagePathIfExists: (...args: string[]) => '';
    hasImage: (...args: string[]) => boolean;
    reloadAll: () => {};
    setAppGroup: (appGroup: string) => void;
}

// This call loads the native module object from the JSI.
export default requireNativeModule<WidgetModule>('Widget');

File: modules/widget/android/src/main/java/com/aoe2companion/widget/WidgetModule.kt
package com.aoe2companion.widget

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.net.URL

class WidgetModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("Widget")

        Events("onChange")

        Function("setAppGroup") {
            "Hello world! 👋"
        }

        Function("reloadAll") {
            "Hello world! 👋"
        }

        Function("setItem") {
            "Hello world! 👋"
        }

        Function("hasImage") {
            "Hello world! 👋"
        }

        Function("setImage") {
            "Hello world! 👋"
        }

        Function("getItem") {
            "Hello world! 👋"
        }

        Function("getAppGroupPath") {
            "Hello world! 👋"
        }
    }
}

class LiveActivityModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("LiveActivity")

        Events("onTokenChanged", "onActivityStarted", "onTest")

        Function("enable") {
            "Hello world! 👋"
        }

    }
}
